operator/internal/lifecycle/v2_ownership.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package lifecycle
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	redpandav1alpha2 "github.com/redpanda-data/redpanda-operator/operator/api/redpanda/v1alpha2"
+)
+
+// V2OwnershipResolver represents an ownership resolver for v2 clusters.
+type V2OwnershipResolver struct {
+	operatorLabel  string
+	ownerLabel     string
+	namespaceLabel string
+}
+
+var _ OwnershipResolver[redpandav1alpha2.Redpanda, *redpandav1alpha2.Redpanda] = (*V2OwnershipResolver)(nil)
+
+// NewV2OwnershipResolver returns a V2OwnershipResolver.
+func NewV2OwnershipResolver() *V2OwnershipResolver {
+	return &V2OwnershipResolver{
+		operatorLabel:  defaultOperatorLabel,
+		ownerLabel:     defaultOwnerLabel,
+		namespaceLabel: defaultNamespaceLabel,
+	}
+}
+
+// AddLabels returns the labels to add to all resources associated with a
+// v2 cluster.
+func (m *V2OwnershipResolver) AddLabels(cluster *redpandav1alpha2.Redpanda) map[string]string {
+	return map[string]string{
+		m.namespaceLabel: cluster.GetNamespace(),
+		m.ownerLabel:     cluster.GetName(),
+		m.operatorLabel:  "v2",
+		// we add these for backwards compatibility for the time being
+		fluxNameLabel:      cluster.Name,
+		fluxNamespaceLabel: cluster.Namespace,
+	}
+}
+
+// GetOwnerLabels returns the labels that can identify a resource belonging
+// to a given cluster.
+func (m *V2OwnershipResolver) GetOwnerLabels(cluster *redpandav1alpha2.Redpanda) map[string]string {
+	return map[string]string{
+		fluxNameLabel:      cluster.Name,
```

Comment:
I think we might want to diverge from flux labels ASAP so as to not "confuse" any deployed flux controllers.

Reply:
So, see above WRT these labels -- let me know if you have any suggestions to handle old resources where this is the _only_ way of determining definitive object ownership since we can't always use ownerRefs.